app/src/main/java/zapsolutions/zap/util/BlockExplorer.java

```diff
@@ -6,123 +6,143 @@
 import android.content.Intent;
 import android.net.Uri;
 import zapsolutions.zap.R;
+import zapsolutions.zap.interfaces.UserGuardianInterface;
 
 /**
  * This class allows showing details of On-Chain transactions and addresses using
  * 3rd Party services (Block Explorers)
  */
-public class BlockExplorer {
+public class BlockExplorer implements UserGuardianInterface {
 
+    private String mUrl;
+    private boolean mSupported;
+    private Context mContext;
+
+
+    private void unsupportedNetwork(String explorer, String network, Context ctx) {
```

Comment:
I would move the private methods of the class to the bottom and the public to the top 👍